network/transparent_endpointclient_linux.go

```diff
@@ -64,33 +66,57 @@ func setArpProxy(ifName string) error {
 }
 
 func (client *TransparentEndpointClient) AddEndpoints(epInfo *EndpointInfo) error {
-	if _, err := net.InterfaceByName(client.hostVethName); err == nil {
+	if _, err := client.netioshim.GetNetworkInterfaceByName(client.hostVethName); err == nil {
 		log.Printf("Deleting old host veth %v", client.hostVethName)
 		if err = client.netlink.DeleteLink(client.hostVethName); err != nil {
 			log.Printf("[net] Failed to delete old hostveth %v: %v.", client.hostVethName, err)
 			return newErrorTransparentEndpointClient(err.Error())
 		}
 	}
 
+	primaryIf, err := client.netioshim.GetNetworkInterfaceByName(client.hostPrimaryIfName)
+	if err != nil {
+		return newErrorTransparentEndpointClient(err.Error())
```

Comment:
Can we not continue if this fails ? we can just ignore setting MTU with a warning message. I am thinking setting MTU is a luxury than necessity and we should not fail the veth pair setup because we could not setup MTU, please let me know if my understanding is wrong ?

Reply:
ya i agree. this feature not needed for all customers. also want to make sure we are not breaking existing behaviour. let me just log it and not return error